vdoctl/cmd/drivers.go

```diff
@@ -0,0 +1,417 @@
+		if err != nil {
+			panic(err)
+		}
+		fmt.Println("Thanks For configuring VDO. The drivers should be installed/configured soon")
+
+	},
+}
+
+func init() {
+	configureCmd.AddCommand(driversCmd)
+
+}
+
+func createSecret(cl client.Client, ctx context.Context, cred credentials, driver string) (v1.Secret, error) {
+
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-creds-%s", cred.vcIp, driver),
+			Namespace: KubeSystemNamespace,
+		},
+		Type: secretType,
+
+		Data: map[string][]byte{
+			"username": []byte(cred.username),
+			"password": []byte(cred.password),
+		},
+	}
+
+	err := cl.Create(ctx, &secret, &runtimeclient.CreateOptions{})
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return secret, nil
+		}
+	}
+	return secret, err
+
+}
+
+func createVsphereCloudConfig(cl client.Client, ctx context.Context, cred credentials, secretName string, driver string) (v1alpha1.VsphereCloudConfig, error) {
+	vcc := &v1alpha1.VsphereCloudConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", cred.vcIp, driver),
+			Namespace: VdoNamespace,
+		},
+		Spec: v1alpha1.VsphereCloudConfigSpec{
+			VcIP:        cred.vcIp,
+			Insecure:    cred.insecure,
+			Credentials: secretName,
+			Thumbprint:  cred.thumbprint,
+			DataCenters: cred.datacenters,
+		},
+		Status: v1alpha1.VsphereCloudConfigStatus{},
+	}
+
+	err := cl.Create(ctx, vcc, &runtimeclient.CreateOptions{})
+
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return *vcc, nil
+		}
+		fmt.Printf("Prompt failed %v\n", err)
+		os.Exit(1)
+
+	}
+	return *vcc, err
+}
+
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&v1alpha1.VsphereCloudConfig{},
+		&v1alpha1.VsphereCloudConfigList{},
+		&v1alpha1.VDOConfig{},
+	)
+	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
+
+	return nil
+}
+
+func createVDOConfig(cl client.Client, ctx context.Context, cpi credentials, csi credentials) error {
+
+	vdoConfig := &v1alpha1.VDOConfig{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vdoConfigName + randstr.Hex(4),
+			Namespace: VdoNamespace,
+		},
+		Spec: v1alpha1.VDOConfigSpec{
+			StorageProvider: v1alpha1.StorageProviderConfig{
+				VsphereCloudConfig:  csi.vsphereCloudConfig,
+				ClusterDistribution: ClusterDistribution,
+				FileVolumes: v1alpha1.FileVolume{
+					VSanDataStoreUrl: csi.vSANDataStoresUrl,
+					NetPermissions:   csi.netPermissions,
+				},
+			},
+		},
+	}
+
+	if len(cpi.vsphereCloudConfigs) > 0 {
+		vdoConfig.Spec.CloudProvider = v1alpha1.CloudProviderConfig{
+			VsphereCloudConfigs: cpi.vsphereCloudConfigs,
+			Topology:            cpi.topology,
+		}
+	}
+
+	err := cl.Create(ctx, vdoConfig, &runtimeclient.CreateOptions{})
+
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return nil
+		}
+		fmt.Printf("Prompt failed %v\n", err)
+		os.Exit(1)
+
+	}
+	return err
+}
+
+func buildConfig() (*rest.Config, error) {
```

Comment:
Can we move this to some common place like root.go, as this will be used to across vdoctl to create resources.
/cc @jvrahav

Reply:
This has been handled in the code in root.go https://github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/blob/main/vdoctl/cmd/root.go#L120 this function sets the global variables for the clients

Reply:
@jvrahav That makes sense, it can be utilised by all the commands to get/create/update resources in k8s.
@aggarwalta you can update the code to use the K8sClient

Reply:
```
         err := CreateNamespace(K8sClient, ctx)
		if err != nil {
			panic(err)
		}

		err = CreateConfigMap(filePath, K8sClient, ctx)
		if err != nil {
			panic(err)
		}
```